docker/Dockerfile

```diff
@@ -1,7 +1,70 @@
-FROM --platform=linux/amd64 node:22-alpine
-WORKDIR /app
+FROM --platform=linux/amd64 node:22-alpine as build
+WORKDIR /build
 COPY . .
-RUN apk add --update python3 make g++ curl bash\
+# we need coreutils for proper ls command
+RUN apk add --update coreutils python3 make g++ curl bash gawk jq\
+   && rm -rf /var/cache/apk/*
+# RUN yarn install
+# RUN yarn build
+
+## Export libraries into archives
+WORKDIR '/build/packages/lib'
+# pack libraries into archives. they are copied to and used in the delivery service image.
+# we ignore the billboard-api library because it is not properly connected to our monorepo
+RUN mkdir /library_archives
+RUN for lib in `ls -I billboard-api`; do cd $lib; yarn pack --filename /library_archives/$lib.tgz; cd ..;  done
+
+## prepare dependency operations for all packages
+WORKDIR '/build/packages'
+# several steps are performed here for each package individually (but not for the libraries):
+# 1. extract the information which libraries are needed for this package and store it in required_libraries
+# 2. remove all dependencied referencing "workspace" in the monorepo from package.json
+# 3. update the package.json with the resolutions from the docker/resolutions.json file
+RUN for package in `ls -I lib`;\
```

Comment:
Can you add a tiny comment above each bash line to explain what it does? That would increase the readability a lot

Reply:
done 23e9b1a